src/Database/LSMTree/Internal/Run.hs

```diff
@@ -40,6 +40,7 @@ module Database.LSMTree.Internal.Run (
   , Run (..)
   , addReference
   , removeReference
+  , close
```

Comment:
This function is potentially dangerous, since it could close a run that others still have references to. If we export it, maybe it should be called `unsafeClose`. On the other hand, I don't think you really need it. Usually  `removeReference` should be enough, e.g. to clean up  a run you just created in the tests.

Reply:
Using `removeReference` now